### PR TITLE
[CI] Create a lane to update StreamChat dependency

### PR DESCRIPTION
**Pull request by testableapple**
### 🔗 Issue Link

Resolve https://linear.app/stream/issue/IOS-599

### 🎯 Goal

- Create a lane for bumping the StreamChat dependency

### 🧪 Testing

- Test: https://github.com/GetStream/stream-chat-swiftui/pull/686/files

### CODE REVIEW

File: fastlane/Fastfile [hunk 1]
```diff
@@ -146,6 +146,40 @@ private_lane :appstore_api_key do
...
+
+    file = 'StreamChatSwiftUI.xcodeproj/project.pbxproj'
+    content = File.read(file)
+    if content.include?("minimumVersion = #{current_stream_chat_version}")
+      File.write(file, content.gsub("minimumVersion = #{current_stream_chat_version}", "minimumVersion = #{options[:version]}"))
+    elsif content.include?('branch = develop')
+      File.write(file, content.gsub('kind = branch', "minimumVersion = #{options[:version]}").gsub('branch = develop', 'kind = upToNextMajorVersion'))
+    else
+      UI.user_error!("Something went wrong after trying to modify #{file}.")
+    end
```
Comment (laevandus) [inline fastlane/Fastfile:172]: Ideally we should check for stream chat since there are multiple dependencies. Not sure if it is a big deal.

Comment (testableapple) [reply fastlane/Fastfile:172]: Yeah, but it actually makes things much more complex. A chance that we will hit a different dependency with the same key `minimumVersion = #{current_stream_chat_version}` is very low. We also do not use other dependencies from `develop` ever.

Comment (laevandus) [reply fastlane/Fastfile:172]: Yeah, probably fine. We can improve it if really need to. Makes sense to not complicate this too much.

**Review by nuno-vieira — APPROVED** [review]
LGTM! ✅

**Pull request merged**